R/helpers-ppc.R

```diff
@@ -135,136 +172,103 @@ validate_x <- function(x = NULL, y, unique_x = FALSE) {
 }
 
 
-#' Convert yrep matrix into a molten data frame
+# Internals for grouped plots ---------------------------------------------
+
+#' Modify a call to a `_grouped` function to a call to the ungrouped version
+#' @param fn The new function to call (a string).
+#' @param call The original call (from `match.call(expand.dots = FALSE)`).
+#' @return The new unevaluated call, with additional argument
+#'   `called_from_internal=TRUE` which can be detected by the function to be
+#'   called so it knows not to warn about the `group` and `facet_args` arguments.
+#' @noRd
+ungroup_call <- function(fn, call) {
```

Comment:
Huh, this is quite the strategy. The sight of it makes me nervous but it does yield shorter code.

Reply:
Yeah, it's basically what I came up with to avoid duplicating code in both the grouped and ungrouped functions. I first proposed it in [this comment](https://github.com/stan-dev/bayesplot/issues/151#issuecomment-496323557) to which you rightfully [responded](https://github.com/stan-dev/bayesplot/issues/151#issuecomment-496885009) that it's "pretty tricky stuff" (which it is), but then it sounded like you eventually [agreed](https://github.com/stan-dev/bayesplot/issues/151#issuecomment-496999158) with the approach. 

I'm not opposed to a different approach if you have a better idea (there are probably better ideas out there). That would probably mean starting from scratch on this but I guess that's ok if there's a much better path.

Reply:
I guess we should finish having this conversation before you go down the rabbit hole of reviewing the whole thing, just in case we decide to change the approach.

Reply:
Ha, it's funny that past me had the same response. I don't want to rehash anything, so let's keep it. It's indirect enough that it can be refactored out later if needed.

Reply:
Ok cool. Yeah we can always refactor in the future if we want.